src/control/lib/helper/handler.go

```diff
@@ -0,0 +1,65 @@
+//
+// (C) Copyright 2020 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package helper
+
+import (
+	"encoding/json"
+
+	"github.com/daos-stack/daos/src/control/fault"
+	"github.com/daos-stack/daos/src/control/logging"
+	"github.com/daos-stack/daos/src/control/pbin"
+	"github.com/pkg/errors"
+)
+
+// RequestHandler is an interface that handles a pbin.Request.
+type RequestHandler interface {
+	Handle(logging.Logger, *pbin.Request) *pbin.Response
+}
+
+// NewResponseWithError creates a new pbin.Response indicating a failure.
+func NewResponseWithError(err error) *pbin.Response {
+	f, ok := errors.Cause(err).(*fault.Fault)
+	if !ok {
+		f = pbin.PrivilegedHelperRequestFailed(err.Error())
+	}
+	res := &pbin.Response{
+		Error:   f,
+		Payload: json.RawMessage([]byte("null")),
+	}
+	return res
```

Comment:
Could just return the struct without the intermediate assignment.

Reply:
Fixed